lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -509,3 +510,82 @@ def process(self, temperature, relative_humidity, pressure):
+                before returning wet bulb temperatures.
+            coord_name_to_integrate (iris.cube.Cube):
+                Name of the coordinate to be integrated.
+            start_point (float or None):
+                Point at which to start the integration.
+                Default is None.
+            end_point (float or None):
+                Point at which to end the integration.
+                Default is None.
+            direction_of_integration (string):
+                Description of the direction in which to integrate.
+                Options are 'positive' or 'negative'.
+                'positive' corresponds to the values within the array
+                increasing as the array index increases.
+                'negative' corresponds to the values within the array
+                decreasing as the array index increases.
+        """
+        self.precision = precision
+        self.coord_name_to_integrate = coord_name_to_integrate
+        self.start_point = start_point
+        self.end_point = end_point
+        self.direction_of_integration = direction_of_integration
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<WetBulbTemperatureIntegral: precision: {}, '
+                  'coord_name_to_integrate: {}, start_point: {}, '
+                  'end_point: {}, direction_of_integration: {}>'.format(
+                      self.precision, self.coord_name_to_integrate,
+                      self.start_point, self.end_point,
+                      self.direction_of_integration))
+        return result
+
+    def process(self, temperature, relative_humidity, pressure):
+        """
+        Calculate the wet bulb temperature integral by firstly calculating
+        the wet bulb temperature from the inputs provided, and then
+        calculating the vertical integral of the wet bulb temperature.
+
+        Args:
+            temperature (iris.cube.Cube):
+                Cube of air temperatures (K).
+            relative_humidity (iris.cube.Cube):
+                Cube of relative humidities (%, converted to fractional).
+            pressure (iris.cube.Cube):
+                Cube of air pressures (Pa).
+
+        Returns:
+            wet_bulb_temperature_integral (iris.cube.Cube):
+                Cube of wet bulb temperature integral (Kelvin-metres).
+        """
+        # Calculate wet-bulb temperature.
+        wet_bulb_temperature = (
+            WetBulbTemperature(precision=self.precision).process(
+                temperature, relative_humidity, pressure))
+        # Integrate.
+        wet_bulb_temperature_integral = (
+            Integration(
+                self.coord_name_to_integrate, start_point=self.start_point,
+                end_point=self.start_point,
+                direction_of_integration=self.direction_of_integration
+                ).process(wet_bulb_temperature))
```

Comment:
I think the initiation of these two child plugins should be in this plugin's __init__ function so that all plugins are initialised at once and all plugins are processed at once.

Reply:
The initiation has been moved to the plugin's `__init__` method.